File: gerado/mprj-gsi-sev/api/src/main/java/br/mp/mprj/sev/api/rest/reports/DashboardController.java
/**
 *  Guick Generate class: https://github.com/wdavilaneto/guick
 *  Author: service-wdavilaneto@redhat.com
 *  This source is free under The Apache Software License, Version 2.0
 *  license url http://www.apache.org/licenses/LICENSE-2.0.txt
 */
package br.mp.mprj.sev.api.rest.reports;

import br.mp.mprj.sev.domain.dto.HistogramDto;
import br.mp.mprj.sev.api.persistence.mybatis.DashboardRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpEntity;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import javax.validation.Valid;
import java.util.List;
import java.util.Map;

/**
 *  CRUD Rest Json 'Controller' for entityIntegranteComissao
 *  Guick Generate class:
 *  https://github.com/wdavilaneto/guick
 *  Author: service-wdavilaneto@redhat.com
 */
@RestController
@RequestMapping(value="/reports/dashboard")
public class DashboardController {

    private static final Logger LOGGER = LoggerFactory.getLogger(DashboardController.class);

    @Resource(name = "dashboardRepository")
    private DashboardRepository dashboardRepository;

    @RequestMapping(value = "/allCount", method = RequestMethod.GET)
    public List<HistogramDto> allCount(){
        return dashboardRepository.allCount();
    }

    @RequestMapping(value = "/getResultadoApuracaoCollectionWithSumCandidatoQtVotos", method = RequestMethod.GET)
    List<HistogramDto> getResultadoApuracaoCollectionWithSumCandidatoQtVotos() {
        return dashboardRepository.getResultadoApuracaoCollectionWithSumCandidatoQtVotos();
    }


    @RequestMapping(value = "/getCandidatoCollectionWithSumEleicaoCriterioDesempate", method = RequestMethod.GET)
    List<HistogramDto> getCandidatoCollectionWithSumEleicaoCriterioDesempate() {
        return dashboardRepository.getCandidatoCollectionWithSumEleicaoCriterioDesempate();
    }


    @RequestMapping(value = "/getEleitorCollectionWithSumEleicaoVersao", method = RequestMethod.GET)
    List<HistogramDto> getEleitorCollectionWithSumEleicaoVersao() {
        return dashboardRepository.getEleitorCollectionWithSumEleicaoVersao();
    }




    @RequestMapping(value = "/getUrnaCollectionWithSumEleicaoVersao", method = RequestMethod.GET)
    List<HistogramDto> getUrnaCollectionWithSumEleicaoVersao() {
        return dashboardRepository.getUrnaCollectionWithSumEleicaoVersao();
    }



}

File: gerado/mprj-gsi-sev/front-end/app/scripts/controllers/dashboards/DashboardController.js
'use strict';

/**
 *  Guick Generate class: https://github.com/wdavilaneto/guick
 *  Author: service-wdavilaneto@redhat.com
 *  This source is free under The Apache Software License, Version 2.0
 *  license url http://www.apache.org/licenses/LICENSE-2.0.txt
 *
 * @ngdoc function
 * @name sevApp.controller:dashboardController
 * @description
 * # dashboardController
 * Controller of the sevApp
 */
angular.module('sevApp').controller('dashboardController', ['$scope', '$stateParams', '$modal', '$location', 'dashboardService', function ($scope, $stateParams , $modal, $location, dashboardService ) {

    $scope.allCount = [];
    $scope.options = dashboardService.getBasicOptions();

    dashboardService.allCount( {}, function (data) {
        $scope.allCount = [{ key: 'Quantidade por Entitdade',values: data }];
    });

    $scope.optionsResultadoApuracaoCollectionWithSumCandidatoQtVotos = dashboardService.getBasicOptions();
    $scope.optionsResultadoApuracaoCollectionWithSumCandidatoQtVotos.chart.xAxis.axisLabel = "Candidato";
    $scope.optionsResultadoApuracaoCollectionWithSumCandidatoQtVotos.chart.yAxis.axisLabel = "Candidato QtVotos";
    dashboardService.getResultadoApuracaoCollectionWithSumCandidatoQtVotos( {}, function (data) {
        $scope.getResultadoApuracaoCollectionWithSumCandidatoQtVotos = [{ key: 'ResultadoApuracaoCollection With Sum of Candidato QtVotos',values: data }];
    });


    $scope.optionsCandidatoCollectionWithSumEleicaoCriterioDesempate = dashboardService.getBasicOptions();
    $scope.optionsCandidatoCollectionWithSumEleicaoCriterioDesempate.chart.xAxis.axisLabel = "Eleicao";
    $scope.optionsCandidatoCollectionWithSumEleicaoCriterioDesempate.chart.yAxis.axisLabel = "Eleicao CriterioDesempate";
    dashboardService.getCandidatoCollectionWithSumEleicaoCriterioDesempate( {}, function (data) {
        $scope.getCandidatoCollectionWithSumEleicaoCriterioDesempate = [{ key: 'CandidatoCollection With Sum of Eleicao CriterioDesempate',values: data }];
    });


    $scope.optionsEleitorCollectionWithSumEleicaoVersao = dashboardService.getBasicOptions();
    $scope.optionsEleitorCollectionWithSumEleicaoVersao.chart.xAxis.axisLabel = "Eleicao";
    $scope.optionsEleitorCollectionWithSumEleicaoVersao.chart.yAxis.axisLabel = "Eleicao Versao";
    dashboardService.getEleitorCollectionWithSumEleicaoVersao( {}, function (data) {
        $scope.getEleitorCollectionWithSumEleicaoVersao = [{ key: 'EleitorCollection With Sum of Eleicao Versao',values: data }];
    });




    $scope.optionsUrnaCollectionWithSumEleicaoVersao = dashboardService.getBasicOptions();
    $scope.optionsUrnaCollectionWithSumEleicaoVersao.chart.xAxis.axisLabel = "Eleicao";
    $scope.optionsUrnaCollectionWithSumEleicaoVersao.chart.yAxis.axisLabel = "Eleicao Versao";
    dashboardService.getUrnaCollectionWithSumEleicaoVersao( {}, function (data) {
        $scope.getUrnaCollectionWithSumEleicaoVersao = [{ key: 'UrnaCollection With Sum of Eleicao Versao',values: data }];
    });



}]);

